source/postprocess/visualization/dynamic_topography.cc

```diff
@@ -75,21 +77,36 @@ namespace aspect
           if (cell->is_locally_owned())
             if (cell->at_boundary())
               {
-                // see if the cell is at the *top* boundary, not just any boundary
-                unsigned int top_face_idx = numbers::invalid_unsigned_int;
-                {
-                  for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
-                    if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
+                // see if the cell is at the *top* or *bottom* boundary, not just any boundary
+                unsigned int face_idx = numbers::invalid_unsigned_int;
+                bool at_upper_surface;
```

Comment:
@tjhei If at_upper_surface is undefined face_idx is also not assigned and we check for that in the next if loop 
 if (face_idx == numbers::invalid_unsigned_int)
      continue
If it's not assigned we continue to the next cell so I don't think there's an issue. 
I will add an Assert statement that addresses your second comment.